### PR TITLE
Adjust models.yml

### PR DESCRIPTION
**Pull request by jsangmeister**
Also set some other fields to `text` which might have longer content.

### CODE REVIEW

**Review by r-peschke — APPROVED** [review]

**Comment by ostcar** [comment]
Please wait before merging this. I would like to have a look. I will do it this weekend

**Comment by ostcar** [comment]
I think the following fields should also be text:
* meeting/welcome_title
* meeting/description
* meeting/motion_preamble
* meeting/motions_export_preamble ???
* meeting/users_pdf_welcometitle
* meeting/assignments_export_preamble
* agenda_item/comment
* speaker/note
* motion_change_recommendation/other_description
* poll/description
* vote/value
* assignment/default_poll_description
* projector_countdown/description

File: global/meta/models.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 # Types:
-#     - Nativ datatypes: string, number, boolean, JSON
+#     - Nativ datatypes: string, text, number, boolean, JSON
```
Comment (ostcar) [inline global/meta/models.yml:3]: Could you add a short explanation what the different between string and text is?

**Review by ostcar — CHANGES_REQUESTED** [review]

**Comment by ostcar** [comment]
_From #1101_
> @ostcar do you really use the comment of the `models.yml` to generate code? I find that highly suspicious, but I can add the comment of course.
No. I hardcoded all possible types: https://github.com/OpenSlides/openslides-autoupdate-service/blob/92bd8f7cf8da951350c0558f9e2af8c888b74951/pkg/datastore/gen_request/main.go#L257-L291

I have to add the `text` type to that list. But I think that it is sensible first to change the specification before changing the code.

> > I think it is very important, that someone checks all of this fields an makes a decision, if they need more then 256 chars.
> 
> I don't think it's such a big problem: If someone tries to set more than 256 characters in a string field, he will just get an error. If enough people do that for a specific field, we can go ahead and just change that to `text` as well.
We don't test OpenSlides with real life data. So if someone finds a field that needs more then 256 characters it could be a big problem for him. It can not be fixed fast since it needs an update in the models.yml.

I think it would be better, that unlimited would be the default string-length. I don't understand why this limitation is necessary.

File: global/meta/models.yml (hunk 1 above)
Comment (r-peschke) [reply global/meta/models.yml:3]: Only in backend: The string is limited to 256 char, text is unlimited

**Comment by ostcar** [comment]
vote/value can not have a restrction. For assignment polls, this value can get big when there are a lot of options. See https://github.com/OpenSlides/OpenSlides/wiki/poll.vote

In most cases, a vote should be less then 100 bytes. But in special cases, the vote should not fail, because the field only accepts some values.

For the (future) vote decrypt service, I calculate with 1kb. But even this could not be enough in some cases.

**Comment by jsangmeister** [comment]
> vote/value can not have a restrction. For assignment polls, this value can get big when there are a lot of options. See https://github.com/OpenSlides/OpenSlides/wiki/poll.vote
> 
> In most cases, a vote should be less then 100 bytes. But in special cases, the vote should not fail, because the field only accepts some values.
> 
> For the (future) vote decrypt service, I calculate with 1kb. But even this could not be enough in some cases.

The `poll.vote` action payload differs from the actual `vote/value` field. As @r-peschke said, `vote/value` only contains the actual value of the single vote and refers to a specific vote from exactly _one_ user to exactly _one_ option in _one_ poll. The payload from `poll.vote` is translated into multiple `vote.create` calls.

**Comment by ostcar** [comment]
Ah. I did not realise this.

Can you tell and explain what the biggest value of the field could theoretically be?

**Comment by jsangmeister** [comment]
> Can you tell and explain what the biggest value of the field could theoretically be?

As far as I know, the value is always restricted to `Y`, `N`, `A`. If we add more poll methods, this could theoretically be enhanced in the future, but I doubt that we'll ever need more than one char.

**Comment by ostcar** [comment]
OK.

**Pull request merged**